### PR TITLE
Add documentation for DependencyContext

### PR DESCRIPTION
**Pull request by MSDN-WhiteKnight**
Related to: #3418

### CODE REVIEW

File: xml/Microsoft.Extensions.DependencyModel/DependencyContext.xml [hunk 1]
```diff
@@ -228,8 +257,8 @@
         <ReturnType>Microsoft.Extensions.DependencyModel.TargetInfo</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>Gets the application's target runtime information.</summary>
```
Comment (gewarren) [inline xml/Microsoft.Extensions.DependencyModel/DependencyContext.xml:260]:
```suggestion
        <summary>Gets information about the application's target runtime.</summary>
```

File: xml/Microsoft.Extensions.DependencyModel/DependencyContext.xml [hunk 2]
```diff
@@ -40,12 +53,12 @@
...
-        <param name="compileLibraries">To be added.</param>
-        <param name="runtimeLibraries">To be added.</param>
-        <param name="runtimeGraph">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="target">Target runtime information.</param>
+        <param name="compilationOptions">Compilation options.</param>
+        <param name="compileLibraries">Compilation libraries.</param>
+        <param name="runtimeLibraries">Runtime libraries.</param>
+        <param name="runtimeGraph">Runtime identifiers graph.</param>
+        <summary>Initializes a new dependency context.</summary>
```
Comment (gewarren) [inline xml/Microsoft.Extensions.DependencyModel/DependencyContext.xml:61]:
```suggestion
        <summary>Initializes a new instance of the <see cref="T:Microsoft.Extensions.DependencyModel.DependencyContext" /> class.</summary>
```

**Review by gewarren — COMMENTED** [review]
Thanks for filling this in @MSDN-WhiteKnight

**Review by eerhardt — APPROVED** [review]
Thanks for this @MSDN-WhiteKnight! This looks like a great addition to me.

**Pull request merged**